Melodot milestone 2

DarkingLee opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] This pull request is being made by the same account as the accepted application.
- [x] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [x] In case of acceptance, an invoice must be submitted and the payment will be transferred to the BTC/ETH/fiat account provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/1804

keeganquigley commented:
Thanks for the delivery @DarkingLee and sorry for the delay. Here is my [initial evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/melodot_2_keeganquigley.md). The Docker script is failing on two operating systems for me, wondering if you can look into it. Thanks!

DarkingLee commented:
Hello @keeganquigley, thank you for your detailed evaluation. I have essentially addressed the issues you mentioned in the evaluation.

1. Docker has been fixed and now runs smoothly on both macOS and Ubuntu.
2. macOS can now compile, but unfortunately, it encounters an upstream issue  [paritytech/substrate/#13804](https://github.com/paritytech/substrate/pull/13804) It hasn't actually been resolved: [paritytech/polkadot-sdk/#1755](https://github.com/paritytech/polkadot-sdk/issues/1755) I've tried mixing several different versions, but none of them work. I believe it's not feasible to fix this within this milestone as trying numerous version combinations would require extensive compilation time. We'll tackle this in the next milestone (either find the right version or implement another fix), so I don't have to wait for compilations.
2. The missing `protoc` crate issue has been added to the setup.

keeganquigley commented:
Thanks for the fixes @DarkingLee everything works now and I am happy to pass the milestone. I have updated the evaluation accordingly. Thanks for the great work!
